sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DoNotOverwriteCollectionElements.vb

```diff
@@ -4,214 +4,232 @@ Imports System.Linq
+            dict((0)) = 0       ' Secondary
+            dict.Add((0), 1)    ' Noncompliant
+        End Sub
+
+        Private Sub SameIndexOnArray1(ByVal array() As Integer)
+            array(0) = 0 ' Secondary
+            array(0) = 1 ' Noncompliant
+        End Sub
+
+        Private Sub SameIndexOnArray2(ByVal array() As Integer)
+            array(0) = 1                ' Secondary
+            array(0) = (array(0) + 1)   ' Noncompliant
+        End Sub
+
+        Private Sub SameIndexOnArray3(ByVal obj As CustomIndexerOneArg)
+            obj("foo") = 0
+            obj("foo") = 1 ' Compliant, obj is not a collection
+        End Sub
+
+        Private Sub SameIndexOnArray4(ByVal obj As CustomIndexerMultiArg)
+            obj("s", 1, 1) = 0
+            obj("s", 1, 1) = 1 ' Compliant obj is not a dictionary
+        End Sub
+
+        Private Sub SameIndexOnList(ByVal list As List(Of Integer))
+            list(0) = 0 ' Secondary
+            list(0) = 1 ' Noncompliant
+        End Sub
+
+        Private Sub ListAdd(ByVal list As List(Of String))
+            '#2674 List.Add method should not raise any issue when used with same elements
+            list.Add("MyText")
+            list.Add("MyText")
+        End Sub
+
+        Private Sub ICollectionAdd(list As ICollection(Of String))
+            '#2674 List.Add method should not raise any issue when used with same elements
+            list.Add("MyText")
+            list.Add("MyText")
+        End Sub
+
+        Private Sub SameIndexSpacedOut(ByVal names() As String)
+            names("a") = "a" ' Secondary
+            names("b") = "b"
+            names("a") = "c" ' Noncompliant
+        End Sub
+
+        Private Sub NonSequentialAccessOnSameIndex(ByVal values() As Integer)
+            Dim index As Integer = 0
+            values(0) = 1
+            index = (index + 1)
+            values(0) = 2 ' FN - We only take consecutive element access
+        End Sub
+
+        Private Sub NonConstantConsecutiveIndexAccess(ByVal values() As Integer)
+            Dim index As Integer = 0
+            values(index) = 1 ' Secondary
+            values(index) = 2 ' Noncompliant
+        End Sub
+
+        Private Sub IDictionaryAdd(ByVal dict As IDictionary(Of Integer, Integer))
+            dict.Add(0, 0) ' Secondary
+'           ^^^^^^^^^^^^^^
+            dict.Add(0, 1) ' Noncompliant
+'           ^^^^^^^^^^^^^^
+        End Sub
+
+        Private Sub IDictionaryAddLowercase(ByVal dict As IDictionary(Of Integer, Integer))
+            dict.add(0, 0) ' Secondary
+'           ^^^^^^^^^^^^^^
+            dict.add(0, 1) ' Noncompliant
+'           ^^^^^^^^^^^^^^
+        End Sub
+
+        Private Sub IDictionaryAddMixedCase(ByVal dict As IDictionary(Of Integer, Integer))
+            dict.Add(0, 0) ' Secondary
+'           ^^^^^^^^^^^^^^
+            dict.Add(0, 1) ' Noncompliant
+'           ^^^^^^^^^^^^^^
+        End Sub
```

Comment:
This test used to check whether the rule would also work when casing is different than CamelCase. The formatting here removes the original purpose of the test. It should remain as it was originally:
```
      dict.add(0, 0) ' Secondary
'     ^^^^^^^^^^^^^^
      dict.ADD(0, 1) ' Noncompliant
'     ^^^^^^^^^^^^^^
```

Reply:
I've actually fix this already, but haven't noticed that I got autoformatted again :(